### PR TITLE
feat: implement hover card functionality on scatterchart

### LINKED ISSUE #636: feature: Implement Contributor hover Card feature
### Type of feature

Select the type of feature request, the lowercase should also be the PR prefix.

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

### Current behavior

Is your feature request related to a problem? Please describe.

**Contributors avatar on ScatterChart currently displays contributor and lines touched information on hover**

Screenshot 2022-12-05 at 13 26 20

### Suggested solution

Describe the solution you'd like.

**A new component has been introduced in #607 and should be used to replay the current tooltip as shown in the image below**

Screenshot 2022-12-05 at 13 29 05

### Code of Conduct

By submitting this issue, you agree to follow our [Code of Conduct](https://docs.opensauced.pizza/contributing/code-of-conduct/).

- [x] I agree to follow this project's Code of Conduct

### Contributing Docs

If you plan on contributing code please read - [Contribution Guide](https://docs.opensauced.pizza/contributing/introduction-to-contributing/)

- [x]  I agree to follow this project's Contribution Docs

### PR DESCRIPTION
**Pull request by SunGoldTech**
## What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Description
This PR: 

- Adds the hover card functionality to the scatter chart
- Includes the implementation of the contributor's profile page
- Create a new hook to fetch individual contributor data -`useSingleContributor`
- adds additional `repositories` props several components needed to enable fetching contributors contributions on a created insight dashboard

### Notes: 

- Hit some limitations using the hovercard to navigate to the contributor's profile page due to some limitations from [nivo](https://nivo.rocks/scatterplot/)
- improvised into making individual avatars clickable and routing users to the contributor's profile page
- header bar for PR history and data was temporarily hidden as it needs API updates to populate the data
> A follow up PR will be opened for the header bar implementation

## Related Tickets & Documents
Fixes #636 

## Added tests?

- [ ] 👍 yes
- [x] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [x] 📕 storybook
- [ ] 🙅 no documentation needed

### CODE REVIEW

File: components/molecules/ContributorHoverCard/contributor-hover-card.tsx [hunk 1]
```diff
@@ -1,66 +1,47 @@
...
-              <VscGitPullRequest className="text-[14px]" color="green" />
-            </div>
-            <div>
-              <Text className="!text-light-slate-12">fix: increase paginated response lim </Text>
-            </div>
-            <div className="text-light-slate-11">2mo</div>
-          </div>
-        ))}
+      <div className="">
+        <ContributorTable limit={5} contributor={githubName} topic={topic} repositories={undefined} />
```
Comment (bdougie) [inline components/molecules/ContributorHoverCard/contributor-hover-card.tsx:40]: > repositories={undefined}
Why is this undefined?

**Comment by bdougie** [comment]
FWIW it's worth it. I think the hovercard renders quicker without the contribution blobs. I am noticing a high rate of failures for the repo logos. Do you think we should start adding those to Cloudinary and fetching from there? 

Screen Shot 2022-12-14 at 12 02 24 PM

**Comment by SunGoldTech** [comment]
>Do you think we should start adding those to Cloudinary and fetching from there?

If it doesn’t attract extra cost moving forward, then it's valid and we could have a separate issue to handle this

>Another alternative is using the orange square fallback, similar to the avatar shown for skreddy (pictured).

This should do for now... will be adding the fallback

**Comment by SunGoldTech** [comment]
Hey @bdougie , opened an issue on the limitation in the Nivo chart you mention https://github.com/plouc/nivo/issues/2201

File: components/molecules/ContributorHoverCard/contributor-hover-card.tsx (hunk 1 above)
Comment (SunGoldTech) [reply components/molecules/ContributorHoverCard/contributor-hover-card.tsx:40]: It's optional and not required... I think a better approach is to not pass the props.
Thanks for pointing this out 👍

**Review by brandonroberts — APPROVED** [review]
We can link to Cloudinary in a follow up

**Review by bdougie — CHANGES_REQUESTED** [review]
What's the plan for the orange squares?

**Comment by bdougie** [comment]
> We can link to Cloudinary in a follow-up

We can create a new lib that fetches avatars without rounding the image. https://github.com/open-sauced/insights/pull/643#pullrequestreview-1221046335

Use this as a reference. 
https://github.com/open-sauced/insights/blob/beta/lib/utils/roundedImages.ts

**Comment by SunGoldTech** [comment]
> We can create a new lib that fetches avatars without rounding the image. [#643 (review)](https://github.com/open-sauced/insights/pull/643#pullrequestreview-1221046335)
> 
> Use this as a reference. https://github.com/open-sauced/insights/blob/beta/lib/utils/roundedImages.ts

Should this be implemented in this PR, or moved to a follow-up PR? cc: @bdougie @brandonroberts

**Comment by SunGoldTech** [comment]
> What's the plan for the orange squares?

I've fixed this and the contributors image is now being displayed properly

**Comment by bdougie** [comment]
I set up a dashboard for opensauced and every contributor has zero prs. but when I look at the contributors page there are contributions. The data seems off.

## hover card
Screen Shot 2022-12-18 at 6 50 36 AM

## contributors page
Screen Shot 2022-12-18 at 6 50 55 AM

**Comment by brandonroberts** [comment]
@SunGoldTech The `repositories` aren’t being passed from the `Dashboard` props down to the hover card component, then down to the contributor table so the hook can use them. If we connect those, it will work correctly. We will look at moving `repositories` to a top-level filter in a future PR to avoid the prop drilling.

File: components/molecules/HoverCardWrapper/hover-card-wrapper.tsx [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+  const profile: ContributorsProfileType = {
+    githubAvatar: `https://www.github.com/${username}.png?size=60`,
+    githubName: username,
+    totalPR: contributor[0]?.recent_pr_total
+  };
+
+  return (
+    <>
+      {contributorLoading ? (
```
[the remark this answers is not in the document]
Comment (SunGoldTech) [reply components/molecules/HoverCardWrapper/hover-card-wrapper.tsx]: Thanks for the observation, this has been implemented now

File: pages/user/[username].tsx [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+import { ContributorsProfileType } from "components/molecules/ContributorHoverCard/contributor-hover-card";
+
+const Contributor = () => {
+  const router = useRouter();
+  const { username } = router.query;
+  const contributorLogin = username as string;
+
+  const { data: contributor, isLoading: contributorLoading, isError } = useSingleContributor(contributorLogin);
+
+  const repoList = (contributor[0]?.recent_repo_list || "").split(",").map((repo) => {
```
Comment (brandonroberts) [inline pages/user/[username].tsx:13]: Not a blocker for this PR but we should move this to a utility function

**Review by brandonroberts — APPROVED** [review]
LGTM. Checked public and insight pages

🔥

**Comment by brandonroberts** [comment]
@SunGoldTech merge conflicts need resolving also

**Pull request merged**